src/openms/source/PROCESSING/CENTROIDING/PeakPickerIM.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2002-present, The OpenMS Team -- EKU Tuebingen, ETH Zurich, and FU Berlin
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// --------------------------------------------------------------------------
+// $Author: Timo Sachsenberg $
+// $Maintainer: Timo Sachsenberg $
+// -------------------------------------------------------------------------------------------------------------------------------------------
+
+#include <OpenMS/PROCESSING/CENTROIDING/PeakPickerIM.h>
+
+namespace OpenMS
+{
+    void PeakPickerIM::pickIMTraces(MSSpectrum& spectrum)
+    {        
+        // determine IM format of spectrum
+        IMFormat format = IMTypes::determineIMFormat(spectrum);
+        switch (format)
```

Comment:
Should there be IMFormat::Unknown as well for consistency?

Reply:
The result of determineIMFormat is currently never UNKNOWN but I could  handle it in the switch as well to make this more explicit.